mongo/datadog_checks/mongo/mongo.py

```diff
@@ -879,6 +879,11 @@ def total_seconds(td):
 
         status['stats'] = db.command('dbstats')
         dbstats = {db_name: {'stats': status['stats']}}
+        try:
+            mongo_version = cli.server_info().get('version', '0.0')
+            self.set_metadata('version', mongo_version)
+        except Exception:
+            self.log.exception("Error when collecting the version from the mongo server.")
```

Comment:
`set_metadata` will never raise an exception

Reply:
No but `cli.server_info` potentially

Reply:
In that case, `mongo_version` will be undefined